src/python/pants/backend/python/dependency_inference/parse_python_dependencies_test.py

```diff
@@ -252,6 +252,64 @@ def test_try_except(rule_runner: RuleRunner) -> None:
     )
 
 
+def test_with_contextlib_suppress(rule_runner: RuleRunner) -> None:
+    content = dedent(
+        """\
+        # standard contextlib.suppress
+        with contextlib.suppress(ImportError):
+            import weak0
+
+        # ensure that we reset the weaken status
+        import strong0
+
+        # We should respect the intention of any function that is obviously suppressing ImportErrors
+        with suppress(ImportError):
+            import weak1
+
+        # We should not weaken because of other suppressions
+        with contextlib.suppress(NameError):
+            import strong1
+
+        # Ensure that we preserve the stack of weakens
+        with suppress(ImportError):
+            import weak2_0
+            with suppress(ImportError):
+                import weak2_1
+            import weak2_2
+
+        # Ensure that we preserve the stack of weakens with `try-except`
+        with suppress(ImportError):
+            try:
+                import weak3_0
+            except ImportError:
+                import weak3_1
+            import weak3_2
+
```

Comment:
Can you toss a few more use-cases. Namely the other styles of import scraping: string-like imports and `__import__`

Reply:
__import__ is tested separately, but not with strong/weakening